src/Tests/Microsoft.NET.TestFramework/NuGetTransientErrorDetector.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.NET.TestFramework
+{
+    public static class NuGetTransientErrorDetector
+    {
+        private static readonly List<string> _errorSubstrings = new List<string>()
+        {
+            "A connection attempt failed because the connected party did not properly respond after a period of time",
+            "Response status code does not indicate success: 5", // match any 5xx error
+            "An error occurred while sending the request"
```

Comment:
Could we also check if the error looks like it was raised in the `Restore` target, or in the NuGet `.targets` file?

Reply:
Also check if nuget.targets is there